packages/dashboard/src/components/queryEditor/iotSiteWiseQueryEditor/modeledDataStreamQueryEditor/modeledDataStreamExplorer/modeledDataStreamTable/createModeledDataStreamColumnDefinitions.ts

```diff
@@ -14,13 +19,23 @@ export function createModeledDataStreamColumnDefinitions(): TableProps<
     {
       id: 'latestValue',
       header: 'Latest value',
-      cell: ({ latestValue }) => latestValue,
+      cell: ({ latestValue }) => {
+        if (significantDigits > 0 && typeof latestValue === 'number' && !Number.isInteger(latestValue)) {
```

Comment:
The number.ts file should have an isNumeric check

Reply:
Also, I think we should handle the significantDigits being negative check in the round function instead of wrapping all of these calls in an if block

Reply:
Updated the changes, please check it again